src/FsAutoComplete.Core/UntypedAstUtils.fs

```diff
@@ -1882,3 +1882,33 @@ module Printf =
         | _ -> ()
         //debug "%A" idents
         result.ToArray()
+
+module Completion =
+
+  [<RequireQualifiedAccess>]
+  type Context =
+  | StringLiteral
+  | Unknown
+
+  let atPos (pos: Pos, ast: ParsedInput): Context =
+    let visitor =
+      { new SourceCodeServices.AstTraversal.AstVisitorBase<Context>() with
+
+        member x.VisitExpr(path, traverseExpr, defaultTraverse, expr): Context option =
+          if Range.rangeContainsPos expr.Range pos
+          then
+              match expr with
```

Comment:
this is a very simple AST walk right now, you can imagine it getting bigger as we get more complex.